tasks/install_system.yml

```diff
@@ -0,0 +1,75 @@
+- name: Make sure iSCSI device for {{ vm.host }} @{{ vm.org }} is logged in
+  systemd:
+    name: vm_iscsi@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+  when: vm.storage_type|default(xen_vman_default_storage_type) == "iscsi"
+
+- name: Format root storage for {{ vm.host }} @{{ vm.org }}
+  shell: mkfs.{{ vm.filesystem|default(xen_vman_default_filesystem) }} "/dev/disk/by-path/ip-{{ xen_vman_iscsi_server }}-iscsi-{{ xen_vman_iscsi_base_wwn }}:{{ vm.org }}-{{ vm.host }}-lun-0"
+  when: vm.storage_type|default(xen_vman_default_storage_type) == "iscsi"
+
+- name: Make sure VM root is mounted
+  systemd:
+    name: vm_mount@{{ vm.org }}/{{ vm.host }}.service
+    state: started
+
+- name: Install {{ vm.os.type }} {{ vm.os.version }} at {{ vm.host }} @{{ vm.org }}
+  shell: debootstrap --include openssh-server,python {{ vm.os.version }} /mnt/vms/{{ vm.org }}-{{ vm.host }}
+  when: vm.os.type == "ubuntu"
+
+- name: Create current ansible user at {{ vm.host }} @{{ vm.org }}
+  shell: chroot /mnt/vms/{{ vm.org }}-{{ vm.host }} /usr/sbin/useradd -c "Create for inital use by xen_vman" -G "users,sudo" -m "{{ ansible_user }}"
+  when: vm.os.type == "ubuntu"  # This check is needed since different distributions use different default groups for sudo
+
+- name: Create .ssh directory for {{ ansible_user }} at {{ vm.host }} @{{ vm.org }}
+  file:
+    path: "/mnt/vms/{{ vm.org }}-{{ vm.host }}/home/{{ ansible_user }}/.ssh"
+    mode: "u=rwX,g-rwx,o-rwx"
+    state: directory
+
+- name: Deploy current public key for {{ ansible_user }} at new VM {{ vm.host }} @{{ vm.org }}
+  copy:
+    src: "{{ ansible_ssh_private_key_file }}.pub"
+    dest: "/mnt/vms/{{ vm.org }}-{{ vm.host }}/home/{{ ansible_user }}/.ssh/authorized_keys"
+    mode: "u=rw,g-rwx,o-rwx"
+
+# Set the correct owner in chroot, since ansible is not able to specify the uid and guid numberic
+- name: Set correct owner for authorized_keys at {{ vm.host }} @{{ vm.org }}
```

Comment:
:+1:

Reply:
Also, do you see how this would suck if you installed the VM asynchronously?

Reply:
I opened in issue for that, you have do do this completely install_system.yml asynchronously.